internal/affine/colorm.go

```diff
@@ -167,6 +167,164 @@ func (c *ColorM) UnsafeElements() ([]float32, []float32) {
 	return eb, et
 }
 
+func (c *ColorM) det() float32 {
+	if !c.isInited() {
+		return 1
+	}
+
+	m00 := c.body[0]
+	m01 := c.body[1]
+	m02 := c.body[2]
+	m03 := c.body[3]
+	m10 := c.body[4]
+	m11 := c.body[5]
+	m12 := c.body[6]
+	m13 := c.body[7]
+	m20 := c.body[8]
+	m21 := c.body[9]
+	m22 := c.body[10]
+	m23 := c.body[11]
+	m30 := c.body[12]
+	m31 := c.body[13]
+	m32 := c.body[14]
+	m33 := c.body[15]
+
+	b234234 := m22*m33 - m23*m32
+	b134234 := m21*m33 - m23*m31
+	b124234 := m21*m32 - m22*m31
+	b034234 := m20*m33 - m23*m30
+	b024234 := m20*m32 - m22*m30
+	b014234 := m20*m31 - m21*m30
+
+	det := m00*(m11*b234234-m12*b134234+
+		m13*b124234) -
+		m01*(m10*b234234-m12*b034234+
+			m13*b024234) +
+		m02*(m10*b134234-m11*b034234+
+			m13*b014234) -
+		m03*(m10*b124234-m11*b024234+
+			m12*b014234)
+	if det == 0 {
+		return 0
+	}
+	return 1 / det
+}
+
+// IsInvertible returns a boolean value indicating
+// whether the matrix c is invertible or not.
+func (c *ColorM) IsInvertible() bool {
+	return c.det() != 0
+}
+
+// Invert inverts the matrix.
+// If c is not invertible, Invert panics.
+func (c *ColorM) Invert() *ColorM {
+	if !c.isInited() {
+		return nil
+	}
+
+	det := c.det()
+	if det == 0 {
+		panic("ebiten: c is not invertible")
```

Comment:
`affine: c is not invertible`

(There might be some panic messages starting with `ebiten:` in this package, but sorry, they were outdated)

Reply:
I copied from GeoM. My bad!